tarodb/assets_common.go

```diff
@@ -27,6 +27,11 @@ type UpsertAssetStore interface {
 	//  * or use a sort of mix-in type?
 	UpsertGenesisAsset(ctx context.Context, arg GenesisAsset) (int32, error)
 
+	// FetchGenesisID is used to fetch the database ID of asset genesis
+	// information already in the DB.
+	FetchGenesisID(ctx context.Context,
+		arg sqlc.FetchGenesisIDParams) (int32, error)
```

Comment:
IIRC the reason I decided not to use a CTE in the existing insertion queries was that this field is both optional and a reference to data already in the DB - didn't find good info on perf./other impact of adding the CTE to those frequently-used queries if it would be used infrequently.

Reply:
Gotcha, we can revisit at a later point.